taskweaver/memory/shared_memory_entry.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Literal, Optional, Tuple
+
+from taskweaver.memory.type_vars import RoleName
+from taskweaver.utils import create_id
+
+
+@dataclass
+class SharedMemoryEntry:
+    type: Literal["plan", "experience_sub_path"]
+    by: str
+    content: str
+    scope: Literal["round", "conversation"]
+    # The id of the round or conversation
+    scope_id: str
+    aggregation_keys: Tuple[Literal["by", "type", "scope", "scope_id", "id"], ...]
```

Comment:
what are those aggregation keys used for? also maybe could define a variable/type-alias to refer to these types to avoid duplications.

Reply:
Explained [here](https://github.com/microsoft/TaskWeaver/blob/84533b785800f8fe06c1ef9c22a4db972e95cbbe/website/docs/memory.md). It is used to aggregate the entries with the same key value.

Reply:
I can tell how it is currently working, but kind of wondering whether it is necessary to have both the concept of `scope` and `aggergation_keys`. 
Not sure whether i have understood correctly, the attributes of `by`, `scope + scope_id`, and `id` is roughly corresponding to scopes of `Role`, `Round / Conv`, and `Attachment`. The `type` might always be part of the key. Maybe only missing part of natural scope concept is "roles in one round", which could be defined explicitly.

Therefore, there should only need `type`, `scope`, and `content`. The `scope` could be `Conv`, `Round`, `Role`, `RoundRole`, `Attachment` (might not useful). And the scope would define the code range in which the SME is accessible, thus there is no need to specify the key upon query.